src/main/java/by/custom/utilcalculator/domain/TreeMediator.java

```diff
@@ -0,0 +1,16 @@
+package by.custom.utilcalculator.domain;
+
+import by.custom.utilcalculator.domain.constants.Command;
+import by.custom.utilcalculator.domain.constants.steps.Step;
+import by.custom.utilcalculator.domain.tree.CommandTree;
+
+public class TreeMediator {
+    public static boolean validateCommand(final Command requestingCommand, final UserProgress userProgress) {
```

Comment:
I think this method can be moved to `CommandTree`. Additional class `TreeMediator` is redundant.

Reply:
Done